pkg/virt-operator/resource/generate/components/secrets_test.go

```diff
@@ -122,6 +122,51 @@ var _ = Describe("Certificate Management", func() {
 			Expect(count).To(Equal(51))
 		})
 
+		It("should keep newest certificates when truncating", func() {
+			now := time.Now()
+			current := newSelfSignedCert(now.Add(-10*time.Second), now.Add(1*time.Hour))
+			certificates := make([]*tls.Certificate, 0, 60)
+
+			// Create 30 recent valid certificates
```

Comment:
To trigger truncation you need >50 valid certificates, no?
Also, if you add them in random order regarding their validity, you could check afterwards that they were ordered correctly.